### PR TITLE
build updates

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: buildSrc/src/main/groovy/io.micronaut.build.internal.oraclecloud-base.gradle [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+dependencies {
+    implementation libs.jakarta.inject
+    testImplementation mn.micronaut.test.junit5
+    testRuntimeOnly libs.junit.jupiter
+    testAnnotationProcessor mn.micronaut.inject.java
+}
+
+tasks.withType(JavaCompile) {
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.oraclecloud-base.gradle:16]: :x: Should be `tasks.withType(JavaCompile).configureEach`

File: oraclecloud-atp/build.gradle [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
-    api "com.oracle.database.jdbc:ojdbc11"
+dependencies {
     api "com.oracle.oci.sdk:oci-java-sdk-database:$ociVersion"
-    implementation "com.oracle.database.xml:xdb"
-    implementation "com.oracle.database.security:oraclepki"
-    implementation "com.oracle.database.security:osdt_core"
-    implementation "com.oracle.database.security:osdt_cert"
-
-    implementation project(":oraclecloud-sdk")
+    api 'com.oracle.database.jdbc:ojdbc11'
```
Comment (melix) [inline oraclecloud-atp/build.gradle:7]: Any reason some dependencies here are not using the version catalog?

File: oraclecloud-common/build.gradle [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-        exclude module:'groovy-all'
+        exclude group: 'org.javassist', module: 'javassist'
     }
-    testImplementation "io.micronaut:micronaut-runtime"
-    testImplementation "io.micronaut.test:micronaut-test-spock:$micronautTestVersion"
-    testImplementation 'org.mockito:mockito-core:4.1.0'
+    api mn.micronaut.inject
+    compileOnly mn.graal
+    compileOnly mn.micronaut.runtime
+    runtimeOnly 'org.slf4j:jcl-over-slf4j:1.7.32'
```
Comment (melix) [inline oraclecloud-common/build.gradle:12]: Version catalog?

File: oraclecloud-function/build.gradle [hunk 4]
```diff
@@ -1,19 +1,16 @@
...
-    }
-    testImplementation "io.micronaut.test:micronaut-test-spock:$micronautTestVersion"
-    testImplementation "com.oracle.oci.sdk:oci-java-sdk-objectstorage"
+dependencies {
+    annotationProcessor mn.micronaut.graal
+    api libs.fn.api
+    api mn.micronaut.function
+    api projects.oraclecloudCommon
+    compileOnly mn.graal
+    testImplementation 'com.oracle.oci.sdk:oci-java-sdk-objectstorage'
```
Comment (melix) [inline oraclecloud-function/build.gradle:11]: Version catalog?

File: oraclecloud-vault/build.gradle [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
-    implementation "io.projectreactor:reactor-core:$reactorVersion"
-    api 'com.oracle.oci.sdk:oci-java-sdk-vault'
     api 'com.oracle.oci.sdk:oci-java-sdk-secrets'
-
-    testImplementation("org.spockframework:spock-core:${spockVersion}") {
-        exclude module:'groovy-all'
-    }
-    testImplementation "io.micronaut.test:micronaut-test-spock:$micronautTestVersion"
-
+    api 'com.oracle.oci.sdk:oci-java-sdk-vault'
```
Comment (melix) [inline oraclecloud-vault/build.gradle:7]: This can probably be defined in the catalog too

**Review by melix — APPROVED** [review]
Overall looks pretty good. I didn't check it functionally, though. I have left a few comments, mostly looks like there are still some dependencies defined with GAV coordinates instead of the catalog.

**Comment by burtbeckwith** [comment]
Fixes https://github.com/micronaut-projects/micronaut-oracle-cloud/issues/306

**Pull request merged**